File: lemniscate/src/test/java/com/vlad1m1r/lemniscate/base/models/CurveSettingsTest.java
package com.vlad1m1r.lemniscate.base.models;

import android.graphics.Paint;

import com.vlad1m1r.lemniscate.TestConstants;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class CurveSettingsTest {

    private CurveSettings curveSettings;

    @Mock
    Paint paint;

    @Before
    public void setUp() {
        curveSettings = new CurveSettings(paint, null);
    }

    @Test
    public void setStrokeWidth() {
        curveSettings.setStrokeWidth(10);
        assertEquals(10, curveSettings.getStrokeWidth(), TestConstants.DELTA);
        verify(paint).setStrokeWidth(10);
    }

    @Test(expected = IllegalArgumentException.class)
    public void setStrokeWidthException() throws Exception {
        curveSettings.setStrokeWidth(-1);
    }

    @Test
    public void setColor() {
        curveSettings.setColor(123);
        assertEquals(123, curveSettings.getColor());
        verify(paint).setColor(123);
    }
}

File: lemniscate/src/test/java/com/vlad1m1r/lemniscate/base/models/CurveSettingsTest.kt
package com.vlad1m1r.lemniscate.base.models

import android.graphics.Paint
import com.vlad1m1r.lemniscate.TestConstants
import com.vlad1m1r.lemniscate.base.settings.CurveSettings
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class CurveSettingsTest {

    private lateinit var curveSettings: CurveSettings

    @Mock
    lateinit var paint: Paint

    @Before
    fun setUp() {
        val lineLength = LineLength()
        curveSettings = CurveSettings(paint, lineLength)
    }

    @Test
    fun setStrokeWidth() {
        curveSettings.strokeWidth = 10f
        assertEquals(10.0f, curveSettings.strokeWidth, TestConstants.DELTA)
        verify<Paint>(paint).strokeWidth = 10f
    }

    @Test(expected = IllegalArgumentException::class)
    fun setStrokeWidthException() {
        curveSettings.strokeWidth = -1f
    }

    @Test
    fun setColor() {
        curveSettings.color = 123
        assertEquals(123, curveSettings.color.toLong())
        verify<Paint>(paint).color = 123
    }
}